src/Catrobat/Controller/Web/ProgramController.php

```diff
@@ -150,7 +154,28 @@ public function projectAction(Request $request, string $id): Response
     ]);
   }
 
-  /**
+    /**TODO works now I can steal programs and now i need to test it with behat
+     */
+    /**
+     * @Route("/project/{id}/steal", name="steal", methods={"GET"})
+     *
+     * @throws ORMException
+     */
+    public function stealGameAction(Request $request, string $id): Response
+    {
+        /** @var User|null $user */
+        $user = $this->getUser();
+
+        /** @var Program $project */
+        $project = $this->program_manager->find($id);
```

Comment:
What happens if this project does not exist?

Reply:
i`if (null === $user) {
      return new Response('Not Logged IN', 500);
    }`

Reply:
Would this be an API, it would be better to use a fitting response code.
500 is typically only used for an internal server error. In this case, a 401 (unauthorized) would fit way better.

However, since this is just an internal route I would recommend using a redirect. Simply redirect the user to the login page.